File: Prac1/PR1_3/output.js
javascript
function complex_math(n, x, y) {
    // базовые операции
    let add = x + y;
    let sub = x - y;
    let mul = x * y;

    // защита от деления на ноль
    let div = y !== 0 ? x / y : 0;

    // степень
    let power = Math.pow(x, 2) + Math.pow(y, 3);

    // факториал
    function factorial(k) {
        if (k <= 1) {
            return 1;
        }
        return k * factorial(k - 1);
    }

    let fact = factorial(n);

    // фибоначчи
    function fibonacci(k) {
        if (k <= 1) {
            return k;
        }
        return fibonacci(k - 1) + fibonacci(k - 2);
    }

    let fib = fibonacci(n);

    // итоговая формула
    let result = add + sub + mul + div + power + fact + fib;

    return result;
}

File: Prac1/PR1_3/output.java
import java.util.function.IntUnaryOperator;

public class ComplexMath {
    public static int complex_math(int n, int x, int y) {
        // базовые операции
        int add = x + y;
        int sub = x - y;
        int mul = x * y;

        // защита от деления на ноль
        double div = y != 0 ? (double) x / y : 0;

        // степень
        double power = Math.pow(x, 2) + Math.pow(y, 3);

        // факториал
        IntUnaryOperator factorial = new IntUnaryOperator() {
            public int applyAsInt(int k) {
                if (k <= 1)
                    return 1;
                return k * applyAsInt(k - 1);
            }
        };
        int fact = factorial.applyAsInt(n);

        // фибоначчи
        IntUnaryOperator fibonacci = new IntUnaryOperator() {
            public int applyAsInt(int k) {
                if (k <= 1)
                    return k;
                return applyAsInt(k - 1) + applyAsInt(k - 2);
            }
        };
        int fib = fibonacci.applyAsInt(n);

        // итоговая формула
        double result = add + sub + mul + div + power + fact + fib;

        return (int) result;
    }

    public static void main(String[] args) {
        System.out.println(complex_math(5, 3, 2));
    }
}